.github/workflows/docker-unified.yml

```diff
@@ -64,6 +64,18 @@ jobs:
     steps:
       - name: Check out the repo
         uses: hsheth2/sane-checkout-action@v1
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          cache: "pip"
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: "zulu"
+          java-version: 17
+      - name: Run lint on smoke test
```

Comment:
this should not be in the setup job, since it'll block the rest of the docker builds

Reply:
That is actually the intention. Lint should cause rest of CI to simply not run. This causes anyone making changes in smoke tests to fail fast. Otherwise CI will wait for docker builds which takes a while before failing.

Reply:
I'm not sure I agree with that - this will add a few minutes of wall clock time to every docker-unified run, which is already the slowest part of our CI. Lint issues are going to be relatively rare in smoke test, so I don't think it makes sense to slow down our CI significantly to avoid a small number of wasted docker builds

Instead the lint should be a separate job so that the docker builds can run in parallel, and we can use something like this https://github.com/orgs/community/discussions/38361 to cancel the other workflows